json_serializable/lib/src/type_helpers/enum_helper.dart

```diff
@@ -20,11 +26,18 @@ class EnumHelper extends TypeHelper {
   String deserialize(DartType targetType, String expression, bool nullable,
       DeserializeContext context) {
     if (targetType is InterfaceType && targetType.element.isEnum) {
+      String wrappedExpression;
```

Comment:
[optional] This type of flow is precisely what the `?:` operator is designed for - it makes it clear that the condition surrounds assignment rather than some other side effects.

`var wrappedExpressions = simpleExpressions.hasMatch(expression) ? expression : '{$expression}';`

Reply:
ack